### PR TITLE
Update document to use Guice constructor injection

### PR DESCRIPTION
**Pull request by elharo**

### CODE REVIEW

File: maven-plugin-tools-annotations/src/site/apt/index.apt [hunk 1]
```diff
@@ -132,6 +110,27 @@ public class MyMojo
...
+
+    private final MyComponent component;
+
+    private final MavenSession session;
+
+    private final MavenProject project;
+
+    private final MojoExecution mojoExecution;
+
+    public MyMojo(MyComponent component, MavenSession session, MavenProject project, MojoExecution mojoExecution) {
```
Comment (slachiewicz) [inline maven-plugin-tools-annotations/src/site/apt/index.apt:127]: @Inject annotation missing

Comment (elharo) [reply maven-plugin-tools-annotations/src/site/apt/index.apt:128]: fixed

**Review by slawekjaranowski — APPROVED** [review]
Should be also updated for 3.x branch?

**Comment by elharo** [comment]
> Should be also updated for 3.x branch?

Do we publish both?

**Pull request merged**